internal/statesync/reactor_test.go

```diff
@@ -193,13 +193,15 @@ func setup(
 		rts.reactor.metrics,
 	)
 
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
 	require.NoError(t, rts.reactor.Start(ctx))
 	require.True(t, rts.reactor.IsRunning())
 
-	t.Cleanup(func() {
-		rts.reactor.Wait()
-		require.False(t, rts.reactor.IsRunning())
-	})
+	t.Cleanup(cancel)
```

Comment:
I guess we could move this to 197 and replace the deferred cancel. Having both is harmless, though.

Reply:
oh, this is clearly a bug, the 197 case shouldn't exist